From 70c0699b88f724e121821b549b2e35d31976d2f8 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Mon, 15 Oct 2018 15:21:55 +0300
Subject: [PATCH 4885/5242] MLK-19936-3: sound: asoc: add rate in machine
 driver hw_params func

commit  4160d0939f6840a7e66d2239d670f545c76565bb from
https://source.codeaurora.org/external/imx/linux-imx.git

Add rate when calling dai sysclk so we can set rate after
we change the pdm_src parrent.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-micfil.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-micfil.c b/sound/soc/fsl/imx-micfil.c
index c4721d7..f137791 100644
--- a/sound/soc/fsl/imx-micfil.c
+++ b/sound/soc/fsl/imx-micfil.c
@@ -59,6 +59,7 @@ static int imx_micfil_hw_params(struct snd_pcm_substream *substream,
 	bool tx = substream->stream == SNDRV_PCM_STREAM_PLAYBACK;
 	struct device *dev = rtd->card->dev;
 	unsigned int fmt = SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_NB_NF;
+	unsigned int rate = params_rate(params);
 	int ret, dir;
 
 	/* For playback the XTOR is slave, and for record is master */
@@ -74,7 +75,7 @@ static int imx_micfil_hw_params(struct snd_pcm_substream *substream,
 
 	/* Specific configurations of DAIs starts from here */
 	ret = snd_soc_dai_set_sysclk(rtd->cpu_dai, 0,
-				     0, dir);
+				     rate, dir);
 	if (ret) {
 		dev_err(dev,
 			"%s: failed to set cpu sysclk: %d\n", __func__,
-- 
1.7.9.5

